lib/scripting/objects/pinball-backend/rom-fetcher.ts

```diff
@@ -0,0 +1,122 @@
+import { GamelistDB } from 'wpc-emu';
+import { logger } from '../../../util/logger';
+
+/**
+ * Functions to fetch a WPC ROM file from VPDB.io
+ */
+
+export function downloadGameEntry(pinmameGameName: string): Promise<LoadedGameEntry> {
+	const gameEntry: undefined | GamelistDB.ClientGameEntry = GamelistDB.getByPinmameName(pinmameGameName);
```

Comment:
gameEntry can be undefined if you request a name that is not in the db.. or do I miss something?

Reply:
Yes but TypeScript automatically assigns the type returned by the function. So if you only put:

```ts
const gameEntry = GamelistDB.getByPinmameName(pinmameGameName);
``` 
and you later do `gameEntry.something` it will complain unless you do it in a block where you assert that `gameEntry` is not undefined or use `gameEntry!.something`.

I find it more readable, and if I need to know the type, I just hover over the variable and the IDE will tell me.